### DISCUSSION TITLE
Custom output-file tar_qarto

### DISCUSSION THREAD

**Discussion by mjrolland** [ropensci/tarchetypes#110] [Help] [answered]
Hi,

I have a quarto report `report.qmd` that evolves over time. I would like to keep a copy of the consecutive exports, and not for them to overwrite the previous. When using the `quarto_render` function, I can use the `output-file` option like this: `output-file = str_c("report_", Sys.Date(), ".html")`.

However, I am not able to pass this option through the `tar_quarto` command...

Thank you in advance for your help.

All the best,
Matthieu

**Comment by wlandau** [comment, marked as answer]
You could put the HTML output under Git / version control and track the version history in GitHub's interface. 

The idea of the `targets` package and other [Make](https://www.gnu.org/software/make/)-like tools is to maintain a single pipeline with newer targets overwriting previous ones. So there is a philosophical difference.

**Reply by mjrolland** [reply]
thank you for your answer, i get it, i was hoping you would _not_ answer that :)

**Comment by ginolhac** [comment]
I suggest you have a look to this slide: https://thomasmock.quarto.pub/reports-presentations/#/render-with-parameters-1
`tar_quarto()` has the `execute_params` arguments that is described to tweak the parameters used and the output file name.

For reference, the code from **Tom Mock**:
``` r
render_fun <- function(penguin){
  quarto::quarto_render(
    input = "penguin-report.qmd",
    execute_params = list(species = penguin),
    output_file = glue::glue("{penguin}-report.html")
  )
}

unique(penguins$species) |>
  as.character() |> 
  purrr::walk(render_fun)
```
